src/components/posts/PostForm.js

```diff
@@ -4,23 +4,50 @@ import { CategoryContext } from "../categories/CategoryProvider"
 
 export const PostForm = (props) => {
 
-    const { addPost, getLastPost } = useContext(PostContext)
+    const { posts, getPosts, addPost, updatePost, getLastPost } = useContext(PostContext)
     const { categories, getCategories } = useContext(CategoryContext)
 
     const [post, setPost] = useState({})
 
+    const editMode = props.match.params.hasOwnProperty("postId")
+
     const handleControlledInputChange = (eve) => {
         const newPost = Object.assign({}, post)
         newPost[eve.target.name] = eve.target.value
         setPost(newPost)
     }
 
+    const getPostInEditMode = () => {
+        if(editMode) {
+            const postId = parseInt(props.match.params.postId)
+            const selectedPost = posts.find(p => p.id === postId) || {}
+            setPost(selectedPost)
+        }
+    }
+
```

Comment:
This totally works but you could also use getPostById to get the post and send less info through http since we have the get_single_post method on server side.